spec.html

```diff
@@ -14425,11 +14470,44 @@ <h1>Runtime Semantics: DestructuringAssignmentEvaluation</h1>
           1. If _iteratorRecord_.[[Done]] is *false*, return ? IteratorClose(_iteratorRecord_, _status_).
           1. Return Completion(_status_).
         </emu-alg>
+        <emu-grammar>ObjectAssignmentPattern : `{` AssignmentRestProperty `}`</emu-grammar>
+        <emu-alg>
+          1. Let _excludedNames_ be a new empty List.
+          1. Return the result of performing RestDestructuringAssignmentEvaluation of _AssignmentRestProperty_ with _value_ and _excludedNames_ as the arguments.
+        </emu-alg>
+
+        <emu-grammar>ObjectAssignmentPattern : `{` AssignmentPropertyList `,` AssignmentRestProperty `}`</emu-grammar>
+        <emu-alg>
+          1. Let _excludedNames_ be the result of performing ? PropertyDestructuringAssignmentEvaluation for _AssignmentPropertyList_ using _value_ as the argument.
+          1. Return the result of performing RestDestructuringAssignmentEvaluation of _AssignmentRestProperty_ with _value_ and _excludedNames_ as the arguments.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-runtime-semantics-destructuringassignmentevaluation-nested-rest-prohibition">
+        <h1>Static Semantics: Early Errors</h1>
+        <emu-grammar>
+          AssignmentRestProperty[Yield, Await] :
+            `...` DestructuringAssignmentTarget[Yield, Await]
+        </emu-grammar>
+        <ul><li>It is a Syntax Error if |DestructuringAssignmentTarget| is an |ArrayLiteral| or an |ObjectLiteral|.</li></ul>
+      </emu-clause>
+
+      <emu-clause id="Rest-RuntimeSemantics-PropertyDestructuringAssignmentEvaluation">
```

Comment:
I'd probably keep these runtime semantics in their current `<emu-clause>`, because it's not required that all semantics in a `<emu-clause>` have the same result type and keeping it in the current clause ensures that any existing links into the spec for these semantics are still valid.